src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -452,6 +452,23 @@ const JupyterFrameManager = ({ onClose, frameRef }) => {
   return null
 }
 
+const PeriodicCookieSetter = ({ namespace, clusterName }) => {
+  const signal = useCancellation()
+
+  const periodicallySetCookie = withErrorReporting('Error setting cookie', async () => {
```

Comment:
We recently created a hook for the 15 minute signout that may simplify this code. Using the hook, this component could look like:

```
usePolling(300000)
withErrorReporting('Error setting cookie', () => Ajax().Jupyter.notebooks(namespace, clusterName).setCookie())
return null
```
This hook hasn't made it into dev yet, but it could be part of this PR

Reply:
I spoke with Brett - this approach would need some more work with an Ajax call